README.md

```diff
@@ -128,15 +165,36 @@ OakMap<K,V> oak = builder.build();
 You are welcome to take a look on the OakMap's [full API](https://github.com/yahoo/Oak/wiki/Full-API).
 OakMap's API implements the ConcurrentNavigableMap interface. For improved performance, it offers additional non-standard zero-copy API methods that are discussed below.
 
-### OakBuffers
-OakMap provides two types of memory buffers: *OakRBuffer* (read-only) and *OakWBuffer* (read and write). These buffers support a similar API for read-only Java ByteBuffers and writable Java ByteBuffers, respectively.
-
-OakMap buffers allow the user direct access to the underlying serialized key-value pairs, without needing to worry about  concurrent accesses and memory management. This access reduces unnecessary copies and deserialization of the underlying mappings.
-Note, however, that since OakMap's get method avoids copying the value and instead returns access to the same underlying memory buffer that compute operations update in-place, the reader may encounter different values -- and even value deletions -- when accessing the buffer returned from get multiple times. This is of course normal behavior for a concurrent map that avoids copying.
-
-An OakRBuffer can represent either a key or a value. The OakRBuffer's user can use the standard interface of a *read-only* ByteBuffer, for example, `int getInt(int index)`, `char getChar(int index)`, `limit()`, etc. Note that ConcurrentModificationException can be thrown as a result of any OakRBuffer method in case the mapping is concurrently deleted.
-
-For backward compatibility with applications that are already based on the use of ByteBuffers, Oak Buffers provide the transform method that atomically applies a transformation to a *read-only* instance of the underlying ByteBuffer. For a more comprehensive code example please refer to the [usage](#usage) section. 
+### Oak Buffers
+Oak use dedicated buffer objects to access off-heap memory. 
+These buffers cannot be instantiated by the user and are always supplied to the user by Oak.
+Their interfaces are:
+ 1. `OakReadBuffer`: scoped, read-only access to the data
+ 2. `OakWriteBuffer`: scoped, read and write access to the data
+ 3. `OakScopeFreeBuffer`: scope-free, read-only access to the data
```

Comment:
I see that you decided to use 3 types of buffers instead of old two. Which is may be good, instead of using "same named" OakReadBuffer in different circumstances... However, if so, I would suggest similar names just with some negation as part of the name. For example: internal-external, restricted-free, bounded-unbounded, closed-open, ...

How about OakReadRestrictBuffer and OakReadFreeBuffer?

Reply:
The way I see it, we escalate the user required proficiency, where higher level of proficiency using more descriptive names, and lower levels use minimalistic naming.
For example: proficiency levels
0. Use only scoped buffers: `Oak{R/W}Buffer`
1. Use scope free buffers: `OakScopeFreeBuffer`
2. Use underlying data: `OakUnsafeDirectBuffer`

According to the logic you suggested in your comment, regular buffers should be called `OakSafeBoundedReadBuffer`.
The internal class names could (and should) be more descriptive as the user is not exposed to them.

Similar naming convention in Java:
ByteBuffer -> DirectBuffer (DirectByteBuffer and HeapByteBuffer are internal)

Note that we might have chosen here one of two approaches:
1. Reduced restrictions added to the name
2. Increased restrictions added to the name
But we don't have unscoped OakWBuffer so (2) wouldn't make sense for us.

Reply:
User with lowest level of proficiency may not use zero copy at all. If zero copy is used then Oak{R/W}Buffer AND OakScopeFreeBuffer need to be used. So I do not see here higher level of proficiency... Both OakScopeFreeBuffer and Oak{R/W}Buffer are the same level...